torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -131,30 +131,29 @@ def validate_args_and_maybe_create_graph_inputs(
                 # future, we can clean this up.
                 tracer.create_graph_input("const")
             new_arg = a
-        elif isinstance(a, TensorVariable):
+        # Weird special case, we probably want to delete it or fold it
+        # into the next case (of `a` being placeable into a graph)
+        elif isinstance(a, AutogradFunctionContextVariable):
             if manually_set_subgraph_inputs:
-                new_proxy = tracer.create_graph_input(a.as_proxy().node.name)
-                example_value = a.as_proxy().node.meta["example_value"]
-                new_arg = wrap_fx_proxy(
-                    tx=tx, proxy=new_proxy, example_value=example_value
-                )
-            else:
-                new_arg = a
-        elif isinstance(a, SymNodeVariable):
+                tracer.create_graph_input(a.as_proxy().node.name)
+            new_arg = a
+        # If `a` can be put into a graph
+        elif a.maybe_fx_node() is not None:
             if manually_set_subgraph_inputs:
-                new_proxy = tracer.create_graph_input(str(a.sym_num.node.expr))
+                node = a.maybe_fx_node()
+                new_proxy = tracer.create_graph_input(node.name)
+                example_value = (
+                    node.meta["example_value"] if "example_value" in node.meta else None
```

Comment:
When "example_value" is None? Is it the SymBool inputs logic in builder.py, which doesn't set the example value

Reply:
I'm not sure, I just put this here just in case it is possible

Reply:
Dynamo should produce ```example_value``` during fake tensor propagation if everything works well 😄.